include/devicetree.h

```diff
@@ -880,6 +880,32 @@
  * @{
  */
 
+/**
+ * @brief Does a devicetree node match a status?
+ *
+ * Example devicetree fragment:
+ *
+ *     n: node {
+ *             compatible = "generic-device";
+ *             status = "okay";
+ *     }
+ *
+ * Example usages which evaluate to 1:
+ *
+ *     DT_NODE_HAS_STATUS(DT_NODELABEL(n), okay)
+ *
+ * Example usages which evaluate to 0:
+ *
+ *     DT_NODE_HAS_STATUS(DT_NODELABEL(n), disabled)
+ *
+ * @param node_id node identifier
+ * @param status status value to check for
+ * @return 1 if the node's status property matches status,
+ *         0 otherwise.
+ */
+#define DT_NODE_HAS_STATUS(node_id, status) \
+	IS_ENABLED(DT_CAT(node_id, _STATUS_##status))
```

Comment:
my approach in #25009 works even when status is a macro that needs to be expanded again, unlike this one

Reply:
Changed to be like your version.